lib/nerves_ssh/options.ex

```diff
@@ -245,7 +245,19 @@ defmodule NervesSSH.Options do
   end
 
   defp authentication_daemon_opts(opts) do
-    [system_dir: to_charlist(opts.system_dir), user_dir: to_charlist(opts.user_dir)]
+    [system_dir: safe_dir(opts.system_dir), user_dir: safe_dir(opts.user_dir)]
+  end
+
+  defp safe_dir(dir) do
+    case File.mkdir_p(dir) do
+      :ok ->
+        to_charlist(dir)
+
+      {:error, err} ->
+        tmp = Path.join("/tmp", dir)
```

Comment:
```suggestion
        tmp = Path.join("/tmp/nerves_ssh", dir)
```

I added `nerves_ssh` to lower the chance of an unfortunate collision. I wasn't sure if `File.mkdir_p/1` needed to be called or if the daemon code would do that.